packages/presets/client/src/index.ts

```diff
@@ -83,6 +83,11 @@ export type ClientPresetConfig = {
          * @description Name of the property that will be added to the `DocumentNode` with the hash of the operation.
          */
         hashPropertyName?: string;
+        /**
+         * @description Algorithm used to generate the hash, could be useful if your server expects something specific (e.g., Apollo Server expects `sha256`).
+         * @default `sha1`
+         */
+        hashAlgorithm?: 'sha1' | 'sha256';
```

Comment:
Crypto's `algorithm` parameter is typed as a `string` rather than a union because it solely depends on algorithms supported by the version of OpenSSL on the platform.

I don't know what is a proper type here, as in, what would be an appropriate subset of algorithms that are mainly supported everywhere. If we do something like `'sha1' | 'sha256' | string`, it will end up simplified as `string` and we'll lose the ability to suggest.

I went down the path of only suggesting the default one `sha1` as well as the "new" one `sha256` that could be required by the server (e.g., Apollo Server). Consumers could still submit pull requests to extend the type here if something else is commonly used.

Thoughts? Open to feedback here!

Reply:
I would suggest typing it as: `hashAlgorithm?: 'sha1' | 'sha256' | (string & {});`

That way you keep the suggestions while still allowing users to step outside of the predefined algorithms.

Reply:
I agree with @ponbac. Good idea and a nice TS trick there!

Reply:
This is clever! Will send a new commit for it.

Reply:
btw, don't forget to update the `generateDocumentHash` function signature as well

Reply:
Done in 79aae0a, thanks for the good suggestions. 🙏